### PR TITLE
vim-patch:8.2.{3611,3613}

### PR DESCRIPTION
**Pull request by f380cedric**
…name

Problem:    Crash when using CTRL-W f without finding a file name.
Solution:   Bail out when the file name length is zero.
https://github.com/vim/vim/commit/615ddd5342b50a6878a907062aa471740bd9a847

### CODE REVIEW

File: src/nvim/normal.c [hunk 1]
```diff
@@ -4476,8 +4476,13 @@ bool get_visual_text(cmdarg_T *cap, char_u **pp, size_t *lenp)
       *pp = ml_get_pos(&VIsual);
       *lenp = (size_t)curwin->w_cursor.col - (size_t)VIsual.col + 1;
     }
-    // Correct the length to include the whole last character.
-    *lenp += (size_t)(utfc_ptr2len(*pp + (*lenp - 1)) - 1);
+    if (**pp == NUL) {
+     *lenp = 0;
```
Comment (zeertzjq) [inline src/nvim/normal.c:4480]:
```suggestion
      *lenp = 0;
```

File: src/nvim/path.c [hunk 2]
```diff
@@ -1682,6 +1682,10 @@ char_u *find_file_name_in_path(char_u *ptr, size_t len, int options, long count,
   char_u *file_name;
   char_u *tofree = NULL;
 
+  if (len == 0) {
+   return NULL;
```
Comment (zeertzjq) [inline src/nvim/path.c:1686]:
```suggestion
    return NULL;
```

**Comment by f380cedric** [comment]
@vigoux, linked to CVE-2021-3973

**Review by vigoux — APPROVED** [review]

File: src/nvim/normal.c [hunk 3]
```diff
@@ -4476,8 +4476,13 @@ bool get_visual_text(cmdarg_T *cap, char_u **pp, size_t *lenp)
       *pp = ml_get_pos(&VIsual);
       *lenp = (size_t)curwin->w_cursor.col - (size_t)VIsual.col + 1;
     }
-    // Correct the length to include the whole last character.
-    *lenp += (size_t)(utfc_ptr2len(*pp + (*lenp - 1)) - 1);
+    if (**pp == NUL) {
+      *lenp = 0;
+    }
+    if (*lenp > 0) {
+      // Correct the length to include the whole last character.
```
Comment (zeertzjq) [inline src/nvim/normal.c:4483]: This comment was changed in the original patch.

File: src/nvim/file_search.c [hunk 4]
```diff
@@ -1434,6 +1434,10 @@ char_u *find_file_in_path_option(char_u *ptr, size_t len, int options, int first
   }
 
   if (first == TRUE) {
```
Comment (zeertzjq) [inline src/nvim/file_search.c:1436]: lint

**Comment by zeertzjq** [comment]
I found a potential merge conflict with patch 8.2.3494, and some more changes (including a comment change and a lint change) are needed, so I'll supersede this with #17132.

**Pull request closed without merging**